roles/deploy/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Initialize
+  deploy_helper: path="{{ project_root }}" state=present
+
+- name: Clone project files
+  git: repo="{{ project_git_repo }}"
+       dest="{{ project_source_path }}"
+       version="{{ project_version }}"
+       accept_hostkey=yes
+
+- name: write unfinished file
+  file: path="{{ project_source_path }}/{{ deploy_helper.unfinished_filename }}" state=touch
+
+- name: Copy files to new build dir
+  command: cp -pr {{ project_source_path }} {{ deploy_helper.new_release_path }}
```

Comment:
I'll think about it more carefully in a bit, but could you just clone the project straight into `deploy_helper.new_release_path` instead of cloning to one dir then copying to another? I think f500's version may have cloned to `project_source_path` just to coordinate with `project_deploy_strategy: "synchronize"`, but you've omitted that.